main.go

```diff
@@ -85,9 +93,34 @@ func main() {
 	}
 
 	utils.SetIsOpenShift(mgr.GetConfig())
+	utils.SetIsKnative(mgr.GetConfig())
+
+	if utils.IsKnativeServing() {
+		klog.V(log.I).Infof("Knative serving is installed in the cluster")
+		if knativeServingClient, err = knative.NewKnativeServingClient(mgr.GetConfig()); err != nil {
+			if err != nil {
+				klog.V(log.E).ErrorS(err, "unable to connect with the knative serving system")
+				os.Exit(1)
+			}
+		}
+	}
+
+	if utils.IsKnativeEventing() {
+		klog.V(log.I).Infof("Knative eventing is installed in the cluster")
+		if knativeEventingClient, err = knative.NewKnativeEventingClient(mgr.GetConfig()); err != nil {
+			if err != nil {
+				klog.V(log.E).ErrorS(err, "unable to connect with the knative eventing system")
+				os.Exit(1)
+			}
+		}
+	}
```

Comment:
Here's not the best place to add this. If I install this operator first and then knative, I won't have these clients configured. We have to carry the `RestConfig` to the controllers and create the Knative clients during the reconciliation. 

Alternatively, we can have a factory for these clients to hold them in a singleton pattern. In every reconciliation cycle, we can simply check if Knative is still installed in the cluster and return the client we have.

This accessor then can be used in discovery and we avoid passing all this construct `ProfileExtensions` around. 

Something like `knative.GetClient()` will check if there's an instance ready, if not we create one, cache, and return.

Also, this is not a required state, meaning we can't `os.Exit(1)` if we can't create the client.